File: packages/react-native-module/ios/TonClientModule.m

#import "TonClientModule.h"
#import "tonclient.h"

@implementation TonClientModule

- (dispatch_queue_t)methodQueue
{
    return dispatch_queue_create("react-native.client.ton", DISPATCH_QUEUE_SERIAL);
}
RCT_EXPORT_MODULE()

static NSString* stringFromTon(tc_string_data_t tonString) {
    if (tonString.len == 0 || tonString.content == NULL || *tonString.content == 0) {
        return @"";
    }
    return [[NSString alloc] initWithBytes:tonString.content length:tonString.len encoding:NSUTF8StringEncoding];
}

static tc_string_data_t interopString(NSString* string) {
    tc_string_data_t result = {
        (char*)string.UTF8String,
        [string lengthOfBytesUsingEncoding: NSUTF8StringEncoding]
    };
    return result;
}

static void handleRequest(uint32_t requestId, tc_string_data_t paramsJson, uint32_t responseType, bool finished) {
    NSString* paramsJsonString = stringFromTon(paramsJson);
}


RCT_EXPORT_METHOD(createContext: (NSString*)configJson) {
}

RCT_EXPORT_METHOD(request: (uint32_t) context
                  functionName:(nonnull NSString *)functionName
                  functionParamsJson:(nonnull NSString *)functionParamsJson
                  requestId: (uint32_t) requestId) {
}

RCT_EXPORT_METHOD(destroyContext: (uint32_t)context) {
    tc_destroy_context(context);
}

@end

File: packages/react-native-module/android/src/main/java/ton/TonClientModule.java
package ton;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.Callback;

public class TonClientModule extends ReactContextBaseJavaModule {

    private final ReactApplicationContext reactContext;

    public TonClientModule(final ReactApplicationContext reactContext) {
        super(reactContext);
        this.reactContext = reactContext;
        TonClientJsonInterface.IResponseHandler responseHandler = new TonClientJsonInterface.IResponseHandler() {
            @Override
            public void invoke(long requestId, String paramsJson, int responseType, boolean finished) {

            }
        };
        TonClientJsonInterface.setResponseHandler(responseHandler);
    }

    @Override
    public String getName() {
        return "TonClientModule";
    }

    @ReactMethod
    public void sendRequest(long context, long requestId, String functionName, String functionParamsJson) {
        TonClientJsonInterface.sendRequest(context, requestId, functionName, functionParamsJson);
    }

    @ReactMethod
    public void createContext(String configJson, Callback onResult) {
        onResult.invoke(TonClientJsonInterface.createContext(configJson));
    }

    @ReactMethod
    public void destroyContext(long context) {
        TonClientJsonInterface.destroyContext(context);
    }
}
